File: power/bms_agent.py
class BMSAgent:
    """Simple Battery Management System agent for tests and benchmarks."""

    def __init__(self, temp_limit: float) -> None:
        self.temp_limit = temp_limit
        self.temps: list[float] = []
        self.contactor_open = False

    def ingest(self, temp: float) -> None:
        self.temps.append(temp)
        if self.check_trip():
            self.open_contactor()

    def check_trip(self) -> bool:
        if not self.temps:
            return False
        return self.temps[-1] > self.temp_limit

    def open_contactor(self) -> None:
        self.contactor_open = True
        print("CONTACTOR OPEN")

File: power/bms_agent.cpp
#include <vector>
#include <chrono>
#include <thread>
#include <iostream>

class BMSAgent {
public:
    BMSAgent(double limit) : temp_limit(limit), contactor_open(false) {}

    void ingest(double temp) {
        temps.push_back(temp);
        if (check_trip()) {
            open_contactor();
        }
    }

    bool check_trip() const {
        if (temps.empty()) return false;
        double t = temps.back();
        return t > temp_limit;
    }

    void open_contactor() const {
        std::cout << "CONTACTOR OPEN" << std::endl;
    }

private:
    double temp_limit;
    mutable bool contactor_open;
    std::vector<double> temps;
};
